src/modules/api/user/link/link.controller.ts

```diff
@@ -1,7 +1,27 @@
-import { Controller } from '@nestjs/common';
+import { LinkModule } from './link.module';
+import { Controller, Post, Body, Param, Put } from '@nestjs/common';
 import { LinkService } from './link.service';
+import { LinkDTO } from './link.dto';
+import { response } from 'express';
 
 @Controller('users/:userId/links')
 export class LinkController {
-  constructor(private linkService: LinkService) {}
+  constructor(private readonly linkService: LinkService) {}
+  @Post()
+  async create(@Body() linkDTO: LinkDTO, @Param('userId') userId) {
+    const { name, status, customUrl, redirectsTo, inPool = false } = linkDTO;
+    try {
+      const resp = await this.linkService.create(
+        name,
+        status,
+        customUrl,
+        redirectsTo,
+        inPool,
+        userId,
+      );
```

Comment:
@pinnamanivenkat @VishnuThokala  is this a better way or passing the object, and destructuring it in params is a better way?

Reply:
Not at all. If the parameters exceed more than 3, always destructure them.